src/main/java/org/opentripplanner/client/parameters/TripPlanParameters.java

```diff
@@ -57,6 +74,61 @@ public boolean isArriveBy() {
     }
   }
 
+  public enum OptimizeType {
+    QUICK,
+    SAFE,
+    FLAT,
+    GREENWAYS,
+    TRIANGLE,
+  }
+
+  public static class InputBanned {
```

Comment:
Can you add a few tests for this?

Reply:
I added the planPlaceToPlaceWithBaned test. 
I am unsure if I should separate the agency, route, and trip assert in different tests.